From 9b3476aaf672616d28dbb525e5a6ce39d9a533de Mon Sep 17 00:00:00 2001
From: Maximiliano Sandoval <msandova@gnome.org>
Date: Tue, 18 Mar 2025 18:06:18 +0100
Subject: [PATCH] widget: Explicitly annotate visible's getter

Since
https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/473
g-i will pick up gtk_widget_is_visible as the getter for this property.

See https://gitlab.gnome.org/GNOME/gjs/-/issues/681.
---
 gtk/gtkwidget.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gtk/gtkwidget.c b/gtk/gtkwidget.c
index a565e58b7d3..b6d125e706a 100644
--- a/gtk/gtkwidget.c
+++ b/gtk/gtkwidget.c
@@ -1148,7 +1148,9 @@ gtk_widget_class_init (GtkWidgetClass *klass)
                         -1, G_MAXINT,
                         -1,
                         GTK_PARAM_READWRITE|G_PARAM_EXPLICIT_NOTIFY);
-
+  /**
+   * GtkWidget:visible: (getter get_visible)
+   */
   widget_props[PROP_VISIBLE] =
       g_param_spec_boolean ("visible",
                             P_("Visible"),
-- 
GitLab

From a11310a40b7444b56823da6bbb44b759f2b15cc8 Mon Sep 17 00:00:00 2001
From: Maximiliano Sandoval <msandova@gnome.org>
Date: Thu, 20 Mar 2025 18:44:25 +0100
Subject: [PATCH] widget: Explicitly annotate sensitive's getter

This is a continuation of 91fe0ee63a66a8244b0fe7f68aa06dd25fb127a2.

Since
https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/473
g-i will pick up gtk_widget_is_visible as the getter for this property.

See https://gitlab.gnome.org/GNOME/gjs/-/issues/681.
---
 gtk/gtkwidget.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gtk/gtkwidget.c b/gtk/gtkwidget.c
index b6d125e706a..eb16d2e410c 100644
--- a/gtk/gtkwidget.c
+++ b/gtk/gtkwidget.c
@@ -1158,6 +1158,9 @@ gtk_widget_class_init (GtkWidgetClass *klass)
                             FALSE,
                             GTK_PARAM_READWRITE|G_PARAM_EXPLICIT_NOTIFY);
 
+  /**
+   * GtkWidget:visible: (getter get_sensitive)
+   */
   widget_props[PROP_SENSITIVE] =
       g_param_spec_boolean ("sensitive",
                             P_("Sensitive"),
-- 
GitLab

From 52e2b8555f56dabdb0056e1d375fdacd4081c2df Mon Sep 17 00:00:00 2001
From: Maximiliano Sandoval <msandova@gnome.org>
Date: Thu, 20 Mar 2025 19:01:20 +0100
Subject: [PATCH] action: Annotate sensitive & visible getters

Since
https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/473
g-i will pick up gtk_widget_is_visible as the getter for this property.

See https://gitlab.gnome.org/GNOME/gjs/-/issues/681.
---
 gtk/deprecated/gtkaction.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gtk/deprecated/gtkaction.c b/gtk/deprecated/gtkaction.c
index b89ae778f36..4c1abf2eb4a 100644
--- a/gtk/deprecated/gtkaction.c
+++ b/gtk/deprecated/gtkaction.c
@@ -453,7 +453,7 @@ gtk_action_class_init (GtkActionClass *klass)
 							 TRUE,
 							 GTK_PARAM_READWRITE));
   /**
-   * GtkAction:sensitive:
+   * GtkAction:sensitive: (getter get_sensitive)
    *
    * Whether the action is enabled.
    *
@@ -468,7 +468,7 @@ gtk_action_class_init (GtkActionClass *klass)
 							 TRUE,
 							 GTK_PARAM_READWRITE));
   /**
-   * GtkAction:visible:
+   * GtkAction:visible: (getter get_visible)
    *
    * Whether the action is visible.
    *
-- 
GitLab

From 1076a6ef72e007c9d66ecff11638757a6ec68199 Mon Sep 17 00:00:00 2001
From: Maximiliano Sandoval <msandova@gnome.org>
Date: Fri, 21 Mar 2025 18:34:48 +0100
Subject: [PATCH] widget: Fix typo in docstring's property name

This is a fixup for a11310a40b7444b56823da6bbb44b759f2b15cc8.
---
 gtk/gtkwidget.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk/gtkwidget.c b/gtk/gtkwidget.c
index eb16d2e410c..4aa686253a2 100644
--- a/gtk/gtkwidget.c
+++ b/gtk/gtkwidget.c
@@ -1159,7 +1159,7 @@ gtk_widget_class_init (GtkWidgetClass *klass)
                             GTK_PARAM_READWRITE|G_PARAM_EXPLICIT_NOTIFY);
 
   /**
-   * GtkWidget:visible: (getter get_sensitive)
+   * GtkWidget:sensitive: (getter get_sensitive)
    */
   widget_props[PROP_SENSITIVE] =
       g_param_spec_boolean ("sensitive",
-- 
GitLab


